flink-runtime/src/main/java/org/apache/flink/runtime/util/bash/BashJavaUtils.java

```diff
@@ -92,9 +102,42 @@ public static void main(String[] args) throws FlinkException {
 		JobManagerProcessSpec jobManagerProcessSpec = JobManagerProcessUtils.processSpecFromConfigWithNewOptionToInterpretLegacyHeap(
 			configuration,
 			JobManagerOptions.JVM_HEAP_MEMORY);
+
+		logMasterConfiguration(jobManagerProcessSpec);
+
 		return Collections.singletonList(ProcessMemoryUtils.generateJvmParametersStr(jobManagerProcessSpec));
 	}
 
+	private static void logMasterConfiguration(JobManagerProcessSpec spec) {
+		JobManagerFlinkMemory flinkMemory = spec.getFlinkMemory();
+		LOG.info("Final Master Memory configuration:");
+		LOG.info("  Total Process Memory: {}", spec.getTotalProcessMemorySize().toHumanReadableString());
+		LOG.info("    Total Flink Memory: {}", flinkMemory.getTotalFlinkMemorySize().toHumanReadableString());
+		LOG.info("      Heap:             {}", flinkMemory.getJvmHeapMemorySize().toHumanReadableString());
+		LOG.info("      Off-heap:         {}", flinkMemory.getJvmDirectMemorySize().toHumanReadableString());
+		LOG.info("    JVM Metaspace:      {}", spec.getJvmMetaspaceSize().toHumanReadableString());
+		LOG.info("    JVM Overhead:       {}", spec.getJvmOverheadSize().toHumanReadableString());
+	}
+
+	private static void logTaskExecutorConfiguration(TaskExecutorProcessSpec spec) {
+		TaskExecutorFlinkMemory flinkMemory = spec.getFlinkMemory();
+		MemorySize totalOffHeapMemory = flinkMemory.getManaged().add(flinkMemory.getJvmDirectMemorySize());
+		LOG.info("Final TaskExecutor Memory configuration:");
+		LOG.info("  Total Process Memory:      {}", spec.getTotalProcessMemorySize().toHumanReadableString());
+		LOG.info("    Total Flink Memory:      {}", flinkMemory.getTotalFlinkMemorySize().toHumanReadableString());
+		LOG.info("      Total Heap Memory:     {}", flinkMemory.getJvmHeapMemorySize().toHumanReadableString());
```

Comment:
```suggestion
		LOG.info("      Total JVM Heap Memory:     {}", flinkMemory.getJvmHeapMemorySize().toHumanReadableString());
```

Reply:
I always used `Total` if there are sub components. Not sure if it is necessary, but I'd probably remove either all or none.

Reply:
Then I would prefer to keep both, at least `JVM` to make it more like what we use in docs.